crates/broker/src/provers.rs

```diff
@@ -135,6 +136,12 @@ impl Prover for Bonsai {
         Ok(self.client.upload_img(image_id, image).await.map(|_| ())?)
     }
 
+    async fn has_image(&self, image_id: &str) -> Result<bool, ProverError> {
+        // upload_img will early return true if the image_id already exists
+        // so calling this with empty data is a way to check this
+        Ok(self.client.upload_img(image_id, vec![]).await?)
```

Comment:
What happens though if the image does not exist yet? Would this fail as we are sending an empty Vec, or will it somehow save it?

Reply:
If it doesn't exist then bento/bonsai will abort when it checks that the calculated image_id matches the passed one. So no worry about saving an empty image